include/Common.h

```diff
@@ -16,6 +19,25 @@ namespace Common {
 
     int RandomInt(int min, int max);
 
+    template <typename T>
+    T RandomEvent(std::vector<int> weights, std::vector<T> outcomes) {
+        std::vector <int> events;
+        int k = 0;
+        for (auto it : weights)
+        {
+            for (int i = 0; i < it; i++)
+            {
+                events.push_back(k);
+            }
+            k++;
+        }
+
+        int denominator = std::accumulate(weights.begin(), weights.end(), 0);
+        int rIndex = Common::RandomInt(0, denominator-1);
+        int outIndex = events.at(rIndex);
```

Comment:
* you don't need the `denominator` variable, using `indices.size()` will suffice instead of recomputing sum
* rename `outIndex` to `randomIndex` to be more clear
* combine line 36 & 37: `int randomIndex = indices.at(Common::RandomInt(0, indices.size() - 1))`